solutions/orphaned/usaco-1498.mdx

```diff
@@ -0,0 +1,182 @@
+---
+id: usaco-1498
+source: Gold
+title: The Best Subsequence
+author: Justin Ji
+---
+
+[Official Analysis (C++)](https://usaco.org/current/data/sol_prob2_gold_feb25.html)
+
+<Spoiler title="Hint 1">
+
+Let's assume we have some black box function that counts the number of ones in the range $[l, r]$. Let's consider two
+cases: one where the number of ones in our range is less than $K$, and
+if the number of ones is greater than or equal to $K$. How can we
+handle these two cases?
+
+</Spoiler>
+
+<Spoiler title="Hint 2">
+
+If the number of ones is $\geq K$, then our subsequence can consist of
+all ones. Otherwise, we need to let some amount of zeros into our subsequence.
+
+Let's say the number of ones in our range is $X$. Then, we need to let
+$K-X$ zeros into our range, and we want the first zero to come as far
+back as possible. What does this correspond to in our array?
+
+</Spoiler>
+
+<Spoiler title="Solution">
+
+As denoted in the hints, let $X$ be the number of ones in our range.
+
+### Case 1: $X \geq K$
+
+Our entire subsequence is ones, so the answer is $2^k - 1$.
+
+### Case 2: $X < K$
+
+It's always optimal to take the smallest suffix of $[l, r]$ that
+has $K - X$ zeros. To intuitively understand this, consider the fact that
+we want to maximize the first position where we have a zero in our subsequence. Given that we need to take at least $K-X$ zeros, we at a
+minimum should take the last $K-X$ zeros in our range. Because we are
```

Comment:
there are other ways to maximize the first zero position that don't take all of the last $K-X$ zeros